plugins/webp-uploads/hooks.php

```diff
@@ -77,7 +77,11 @@ function webp_uploads_create_sources_property( array $metadata, int $attachment_
 		$metadata['sources'] = array();
 	}
 
-	if ( empty( $metadata['sources'][ $mime_type ] ) ) {
+	if (
+		! isset( $metadata['sources'][ $mime_type ] ) ||
+		! is_array( $metadata['sources'][ $mime_type ] ) ||
+		0 === count( $metadata['sources'][ $mime_type ] )
+	) {
```

Comment:
Do we need this check here? I know technically this would be the exact replacement for what `empty()` did here, but the array checked here can always be expected to be a non-empty associative array, as it is never set to anything else. So I think the `! isset( ... ) || ! is_array( ... )` would be sufficient here.

The `count()` checks make sense to me mostly when it's an indexed array, i.e. an actual _array_ in the traditional sense of programming.
```suggestion
	if (
		! isset( $metadata['sources'][ $mime_type ] ) ||
		! is_array( $metadata['sources'][ $mime_type ] )
	) {
```